api/src/security/auth-utils.test.ts

```diff
@@ -270,7 +278,161 @@ describe('authorize', function () {
       await auth_utils.authorize({ keycloak_token: 'some token' }, ['abc']);
       expect.fail();
     } catch (actualError) {
-      expect(actualError.message).to.contain('Access Denied');
+      expect(actualError).instanceOf(HTTP403);
+    }
+  });
+
+  it('throws HTTP403 when stubbed getSystemUser throws error', async function () {
+    sinon.stub(auth_utils, 'getSystemUser').callsFake(async function (keycloak_token: object) {
+      if (!keycloak_token || keycloak_token) {
+        throw new Error();
+      }
```

Comment:
nit: you don't need the if statement, you can just throw the error.

Reply:
Yeah, this is a wierd one.  If I don't check ofor the keycloak_token, I get:

  authenticate
[2021-04-29 09:12:46] (warn) (security/auth-utils): authenticate - authorization headers were null or missing  
[2021-04-29 09:12:46] (warn) (security/auth-utils): authenticate - unexpected error - Access Denied 
[object Object] 
[2021-04-29 09:12:46] (warn) (security/auth-utils): authenticate - authorization headers were null or missing  
[2021-04-29 09:12:46] (warn) (security/auth-utils): authenticate - unexpected error - Access Denied 
[object Object] 
[2021-04-29 09:12:46] (warn) (security/auth-utils): authenticate - authorization headers were null or missing  
[2021-04-29 09:12:46] (warn) (security/auth-utils): authenticate - unexpected error - Access Denied 
[object Object] 
    ✓ throws HTTP401 when authorization headers were null or missing
[2021-04-29 09:12:46] (warn) (security/auth-utils): authenticate - authorization header did not have a bearer  
[2021-04-29 09:12:46] (warn) (security/auth-utils): authenticate - unexpected error - Access Denied 
[object Object] 
[2021-04-29 09:12:46] (warn) (security/auth-utils): authenticate - token string was null  
[2021-04-29 09:12:46] (warn) (security/auth-utils): authenticate - unexpected error - Access Denied 
[object Object] 
[2021-04-29 09:12:46] (warn) (security/auth-utils): authenticate - decoded token was null  
[2021-04-29 09:12:46] (warn) (security/auth-utils): authenticate - unexpected error - Access Denied 
[object Object] 
[2021-04-29 09:12:46] (warn) (security/auth-utils): authenticate - decoded token header kid was null  
[2021-04-29 09:12:46] (warn) (security/auth-utils): authenticate - unexpected error - Access Denied 
[object Object] 
[2021-04-29 09:12:47] (warn) (security/auth-utils): authenticate - unexpected error - Unable to find a signing key that matches 'some id' 
SigningKeyNotFoundError: Unable to find a signing key that matches 'some id'

Reply:
we can go over this

Reply:
wonder if I should put a comment explaining why?  otherwise, can we solve it?